web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiClient.java

```diff
@@ -0,0 +1,72 @@
+package dev.langchain4j.web.search.searchapi;
+
+import static com.google.gson.FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES;
+
+import java.io.IOException;
+import java.time.Duration;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import lombok.Builder;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class SearchApiClient {
+	private final SearchApiAPI searchapiAPI;
+	private final String engine;
+	private final String apiKey;
+
+	/* Indicates whether to log requests and their responses. Defaults to false. */
+	private final boolean logRequests;
+
+	private static final Gson GSON = new GsonBuilder().setFieldNamingPolicy(LOWER_CASE_WITH_UNDERSCORES)
+			.setPrettyPrinting().create();
+
+	@Builder
+	public SearchApiClient(final String baseUrl, final String apiKey, final String engine, final Duration timeout,
+			final boolean logRequests) {
+		this.apiKey = apiKey;
+		this.engine = engine;
```

Comment:
If `apiKey`, `baseUrl` or `engine` is null or empty, does it work anyway? If not, you could add:

```suggestion
if (isNullOrBlank(baseUrl)) {
    throw new IllegalArgumentException("baseUrl cannot be null or empty");
}
if (isNullOrBlank(apiKey)) {
    throw new IllegalArgumentException("SearchAPI API Key must be defined. " +
            "It can be generated here: https://www.searchapi.io/api_tokens");
}
if (isNullOrBlank(engine)) {
    throw new IllegalArgumentException("engine parameter must be set");
}
this.apiKey = apiKey;
this.engine = engine;
```